### ISSUE TITLE
re update sapling addon

### ISSUE THREAD

**Issue by meta-androcto** [abpy/improved-sapling-tree-generator#8]
hi, i've been testing your sapling addon again & have been very pleased. I'm happy to commit to blender asap, just a couple of small things.
maybe 1 or 2 lines too long in utils
really would like a way to apply then delete the prune "envelope" object. (from a user perspective it's bad to set up your prune & try to rig & have failure)
That's it, hope your well.
Brendon..

**Comment by abpy** [comment]
I'm not sure if there's a good way to do this. Do you want the envelope object to be deleted when the user is done with the pruning settings or when their done editing the tree?
I can make it so the envelope is only drawn while in prune settings then removed after a setting in another panel is changed.
What do you mean by "try to rig & have failure"? armatures do not work with pruning, I tried to fix it but could not find why it breaks.

**Comment by meta-androcto** [comment]
hi, I've made an updated version here: https://developer.blender.org/T46559
not much really, but armature pruning & animation all seem working nice now. Maybe api changed somewhere or was regression.
Don't worry about pruning, 
This is latest version I have, It includes 3 new presets (including with prune object & original willow) with hanging branches. & new menu layout & fix for object location & whitespace removed.
https://github.com/meta-androcto/blenderpython/raw/master/scripts/addons/add_curve_sapling.7z
One thing i was looking at was the change in attractup function, but still can get good results, just not as smooth as original. let me know when your good as I'm ready to commit & close this as soon as your good. After which create a new task if you have any future updates & I'll see them through.
Thanks.
Brendon.

**Comment by abpy** [comment]
Ok, I'll look at the changes and let you know what I think.

**Comment by abpy** [comment]
hi, I've tested your version and heres what I found.
Armatures and pruning still do not work in 2.76 or 2.77.
I don't think there is any change to attractup.
I like armature and animation in separate panels, I was starting to get too crowded.
I think the prune settings should be after branch growth and before leaves, because pruning is applied to the branches, then leaves and armature are applied to the resulting tree.
